unstructured/metrics/evaluate.py

```diff
@@ -187,3 +173,25 @@ def measure_element_type_accuracy(
     _write_to_file(export_dir, "all-docs-element-type-frequency.tsv", df)
     _write_to_file(export_dir, "aggregate-scores-element-type.tsv", agg_df)
     _display(agg_df)
+
+
+def group_text_extraction_acc(
+    grouping: str, df: pd.DataFrame, export_filename: str
+) -> Tuple[pd.DataFrame, str]:
+    """Aggregate the evaluation score on `grouping` type."""
+    if grouping in ["doctype", "connector"]:
+        grouped_acc = (
+            df.groupby(grouping)
+            .agg({"cct-accuracy": [_mean, _stdev, "count"]})
+            .rename(columns={"_mean": "mean", "_stdev": "stdev"})
+        )
+        grouped_miss = (
+            df.groupby(grouping)
+            .agg({"cct-%missing": [_mean, _stdev, "count"]})
+            .rename(columns={"_mean": "mean", "_stdev": "stdev"})
+        )
+        df = _format_grouping_output(grouped_acc, grouped_miss)
+        export_filename = f"all-{grouping}-agg-cct"
+    else:
+        print("No field to group by. Returning a non-group evaluation.")
+    return df, export_filename
```

Comment:
In general, avoiding nesting improves readability. In this case it could be accomplished with a "guard", like:
```python
    # -- {{explanation of why other grouping types don't count}} --
    if grouping not in ("doctype", "connector"):
        print("No field to group by. Returning a non-group evaluation.")
        return df, export_filename

    grouped_acc = (
        df.groupby(grouping)
        .agg({"cct-accuracy": [_mean, _stdev, "count"]})
        .rename(columns={"_mean": "mean", "_stdev": "stdev"})
    )
    grouped_miss = (
        df.groupby(grouping)
        .agg({"cct-%missing": [_mean, _stdev, "count"]})
        .rename(columns={"_mean": "mean", "_stdev": "stdev"})
    )
    df = _format_grouping_output(grouped_acc, grouped_miss)
    export_filename = f"all-{grouping}-agg-cct"
    return df, export_filename
```

Reply:
Also, why is there a `print()` call? I'm not familiar with this part of the code, so maybe it's part of a CLI or something, but in general a library should not have print statements.

Reply:
The print call is to be shown on CLI terminal if no grouping is done on the df. I'll think of ways to move this elsewhere if possible.

Reply:
It's fine if this is a CLI, I just didn't realize we had that in the library :)